crates/fluvio-cluster/src/cli/shutdown.rs

```diff
@@ -0,0 +1,150 @@
+                "❌",
+                profile_name.italic(),
+            ));
+        }
+
+        Self::kill_local_processes(&self, &pb).await?;
+
+        // Notify the k8s cluster we have removed the spus so that
+        // `fluvio cluster start --local --develop` can be used to restart the cluster
+        let _ = self.remove_custom_objects("spus", DEFAULT_NAMESPACE, None, false, &pb);
+
+        Ok(())
+    }
+
+    async fn kill_local_processes(&self, pb: &ProgressRenderer) -> Result<(), ClusterError> {
+        pb.set_message("Uninstalling fluvio local components");
+
+        let kill_proc = |name: &str, command_args: Option<&[String]>| {
+            let mut sys = System::new();
+            sys.refresh_processes(); // Only load what we need.
+            for process in sys.processes_by_exact_name(name) {
+                if let Some(cmd_args) = command_args {
+                    // First command is the executable so cut that out.
+                    let proc_cmds = &process.cmd()[1..];
+                    if cmd_args.len() > proc_cmds.len() {
+                        continue; // Ignore procs with less command_args than the target.
+                    }
+                    if cmd_args.iter().ne(proc_cmds[..cmd_args.len()].iter()) {
+                        continue; // Ignore procs which don't match.
+                    }
+                }
+                if !process.kill() {
+                    // This will fail if called on a proc running as root, so only log failure.
+                    debug!(
+                        "Sysinto process.kill() returned false. pid: {}, name: {}: user: {:?}",
+                        process.pid(),
+                        process.name(),
+                        process.user_id(),
+                    );
+                }
+            }
+        };
+        kill_proc("fluvio", Some(&["cluster".into(), "run".into()]));
+        kill_proc("fluvio", Some(&["run".into()]));
+        kill_proc("fluvio-run", None);
+
+        // remove monitoring socket
+        match remove_file(SPU_MONITORING_UNIX_SOCKET) {
+            Ok(_) => {
+                pb.println(format!(
+                    "Removed spu monitoring socket: {}",
+                    SPU_MONITORING_UNIX_SOCKET
+                ));
+            }
+            Err(err) => {
+                pb.println(format!(
+                    "SPU monitoring socket  {}, can't be removed: {}",
+                    SPU_MONITORING_UNIX_SOCKET, err
+                ));
+            }
+        }
+
+        pb.println("Uninstalled fluvio local components");
+        pb.finish_and_clear();
+
+        Ok(())
+    }
+
+    fn profile_name(config_file: &ConfigFile) -> String {
+        config_file
+            .config()
+            .current_profile_name()
+            .unwrap()
+            .to_string()
+    }
+
+    /// Remove objects of specified type, namespace
+    fn remove_custom_objects(
+        &self,
```

Comment:
need to keep metadata so that we can restart with existing metadata

Reply:
Without this, we cannot use `cluster start` after `cluster shutdown` to restart SC and SPUs because of the error:

```
./fluvio λ: flvd cluster shutdown
Removed spu monitoring socket: /tmp/fluvio-spu.sock
Uninstalled fluvio local components
./fluvio λ: flvd cluster start --local --develop
📝 Running pre-flight checks
    ✅ Supported helm version 3.9.4+gdbc6d8e is installed
    ✅ Kubectl active cluster rancher-desktop at: https://127.0.0.1:6443 found
    ✅ Supported Kubernetes server 1.24.6+k3s1 found
    ✅ Local Fluvio is not installed
    ✅ Fluvio system charts are installed
🎉 All checks passed!
✅ Local Cluster initialized
🖥️  Trying to connect to SC: localhost:9003 0 seconds elapsed
✅ SC Launched
🖥️  Trying to connect to SC: localhost:9003 0 seconds elapsed
👤 Profile set
Error: 
   0: Fluvio cluster error
   1: Kubernetes client error
   2: MetaStatus {
   2:     api_version: "v1",
   2:     code: Some(
   2:         409,
   2:     ),
   2:     details: Some(
   2:         StatusDetails {
   2:             name: "custom-spu-5001",
   2:             group: Some(
   2:                 "fluvio.infinyon.com",
   2:             ),
   2:             kind: "spus",
   2:             uid: None,
   2:         },
   2:     ),
   2:     kind: "Status",
   2:     message: Some(
   2:         "spus.fluvio.infinyon.com \"custom-spu-5001\" already exists",
   2:     ),
   2:     reason: Some(
   2:         "AlreadyExists",
   2:     ),
   2:     status: FAILURE,
   2: }
``` 

Should I assume the developer will manually restart the SC and SPUs?

I could also add a `cluster restart` command.

Reply:
Actually, `cluster upgrade` can be used